packages/gatsby-source-contentful/src/rich-text.js

```diff
@@ -1,185 +1,69 @@
-      return field
+  // We need the original id from contentful as id attribute
+  // This code can be simplified as soon we properly replicate
+  // Contentful's sys object
+  const traverse = obj => {
+    for (let k in obj) {
+      const value = obj[k]
+      if (
+        value &&
+        value.sys &&
+        value.sys.type === `Link` &&
+        value.sys.contentful_id
+      ) {
+        value.sys.id = value.sys.contentful_id
+        delete value.sys.contentful_id
+      } else if (value && typeof value === `object`) {
+        traverse(value)
+      }
     }
-
-    return getEntryWithFieldLocalesResolved({
-      entry: field,
-      contentTypesById,
-      getField,
-      defaultLocale,
-      resolvedEntryIDs: resolvedEntryIDs.add(key),
-    })
-  }
-
-  if (isAssetReferenceField(field)) {
-    return getAssetWithFieldLocalesResolved({
-      asset: field,
-      getField,
-    })
-  }
-
-  if (Array.isArray(field)) {
-    return field.map(fieldItem =>
-      getFieldWithLocaleResolved({
-        field: fieldItem,
-        contentTypesById,
-        getField,
-        defaultLocale,
-        resolvedEntryIDs,
-      })
-    )
   }
 
-  return field
-}
-
-const getEntryWithFieldLocalesResolved = ({
-  entry,
-  contentTypesById,
-  getField,
-  defaultLocale,
-
-  /**
-   * Keep track of entries we've already resolved, in case two or more entries
-   * have circular references (so as to prevent an infinite loop).
-   */
-  resolvedEntryIDs = new Set(),
-}) => {
-  const contentType = contentTypesById.get(entry.sys.contentType.sys.id)
+  traverse(richText)
 
-  return {
-    ...entry,
-    fields: _.mapValues(entry.fields, (field, fieldName) => {
-      const fieldProps = getFieldProps(contentType, fieldName)
-
-      const fieldValue = fieldProps.localized
-        ? getField(field)
-        : field[defaultLocale]
-
-      return getFieldWithLocaleResolved({
-        field: fieldValue,
-        contentTypesById,
-        getField,
-        defaultLocale,
-        resolvedEntryIDs,
-      })
-    }),
-  }
-}
-
-const getNormalizedRichTextNode = ({
-  node,
-  contentTypesById,
-  getField,
-  defaultLocale,
-}) => {
-  if (isEntryReferenceNode(node)) {
-    return {
-      ...node,
-      data: {
-        ...node.data,
-        target: getEntryWithFieldLocalesResolved({
-          entry: node.data.target,
-          contentTypesById,
-          getField,
-          defaultLocale,
-        }),
+  // Create dummy response so we can use official libraries for resolving the entries
+  const dummyResponse = {
+    items: [
+      {
+        sys: { type: `Entry` },
+        richText,
       },
-    }
-  }
-
-  if (isAssetReferenceNode(node)) {
-    return {
-      ...node,
-      data: {
-        ...node.data,
-        target: getAssetWithFieldLocalesResolved({
-          asset: node.data.target,
-          getField,
+    ],
+    includes: {
+      Entry: references
+        .filter(({ __typename }) => __typename !== `ContentfulAsset`)
```

Comment:
Why not
```suggestion
        .filter(({ __typename }) => __typename === `ContentfulEntry`)
```
? There could potentially be other includes like metadata in the future.